crates/runtime/src/autostart.rs

```diff
@@ -9,17 +9,30 @@ const PERSISTED_FILE_NAME: &str = "autostart_config.msgpack";
 /// Is Auto Start on Boot enabled?
 #[derive(Clone, Serialize, Deserialize, Debug, Default)]
 struct AutostartConfig {
-    enabled: bool,
+    /// Is autostart enabled
+    ///
+    /// To enable autostart, set this to `Some(config)`,
+    /// where `config` is the RuntimeNetworkConfig to apply when starting the runtime.
+    ///
+    /// To disable autostart, set this to `None`.
+    enabled_config: Option<RuntimeNetworkConfig>,
```

Comment:
The autostart config now stores the runtime network config, which should be applied when auto-starting the runtime.